File: frontend/src/services/reviewService.js
import api from './api';

const ReviewService = {
  createReview: async (reviewData) => {
    const response = await api.post('/reviews', reviewData);
    return response.data;
  },

  getReviewsByProductId: async (productId) => {
    const response = await api.get(`/reviews/product/${productId}`);
    return response.data;
  },

  updateReview: async (reviewId, reviewData) => {
    const response = await api.put(`/reviews/${reviewId}`, reviewData);
    return response.data;
  },

  deleteReview: async (reviewId) => {
    await api.delete(`/reviews/${reviewId}`);
  }
};

export default ReviewService;

File: demo/src/main/java/com/smartcommerce/service/serviceInterface/ReviewService.java
package com.smartcommerce.service.serviceInterface;

import com.smartcommerce.dtos.request.UpdateReviewDTO;
import com.smartcommerce.model.Review;
import java.util.List;

public interface ReviewService {
    Review createReview(Review review);
    Review getReviewById(int reviewId);
    List<Review> getReviewsByProductId(int productId);
    List<Review> getReviewsByUserId(int userId);
    List<Review> getAllReviews();
    Review updateReview(int reviewId, UpdateReviewDTO dto);
    void deleteReview(int reviewId);
}
